src/odemis/util/test/spot_test.py

```diff
@@ -24,11 +25,21 @@
 import os
 import scipy.stats
 import unittest
+import warnings
 
 
 TEST_IMAGE_PATH = os.path.dirname(__file__)
 
 
+def ignore_pending_deprecation_warning(func):
+    @functools.wraps(func)
+    def decorated(self, *args, **kwargs):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PendingDeprecationWarning)
+            func(self, *args, **kwargs)
+    return decorated
+
+
```

Comment:
Instead of having all this fancy/complicated code hiding warning in the test cases, it'd be best update all the callers, and not have warnings.